### PR TITLE
Bazel: src/lib/pickles support

### PR DESCRIPTION
**Pull request by mobileink**
This PR adds Bazel support for `src/lib/pickles` and its dependencies.  The targets in this PR will not build until the following PRs are merged:  #7146 #7149 #7150

### CODE REVIEW

File: src/lib/coda_version/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    name = "_Coda_version.cmi",
+    src = "coda_version.mli",
+    ppx = "//bzl/ppx/exe:ppx_version"
+)
+
+########
+genrule(
+    name = "gensrc",
+    stamp = True,
+    srcs = ["coda_version.ml.template"],
```
Comment (mrmr1993) [inline src/lib/coda_version/BUILD.bazel:33]: Did you forget to check this file in?

File: src/lib/coda_version/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    outs = ["coda_version.ml"],
+    cmd = "\n".join([
```
Comment (mrmr1993) [inline src/lib/coda_version/BUILD.bazel:35]: Is it possible to use environment variables somehow instead of an inline bash script?

**Review by mrmr1993 — CHANGES_REQUESTED** [review]

File: src/lib/coda_version/BUILD.bazel (hunk 1 above)
Comment (mobileink) [reply src/lib/coda_version/BUILD.bazel:33]: Yep. will add.

File: src/lib/coda_version/BUILD.bazel (hunk 2 above)
Comment (mobileink) [reply src/lib/coda_version/BUILD.bazel:35]: I don't think so.  This is designed to work with Bazel stamping.  This code is provisional; it works for now, but its a change from the current approach so you folks will need to learn how Bazel stamping works and decide if this is the right way to go.

I've opened an issue on this, https://github.com/MinaProtocol/mina/issues/7163

File: src/lib/coda_version/BUILD.bazel (hunk 1 above)
Comment (mobileink) [reply src/lib/coda_version/BUILD.bazel:33]: done

File: src/lib/coda_version/BUILD.bazel (hunk 2 above)
Comment (mrmr1993) [reply src/lib/coda_version/BUILD.bazel:35]: Great, thanks

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**